test/lib/bamboo/adapters/sendgrid_helper_test.exs

```diff
@@ -0,0 +1,76 @@
+defmodule Bamboo.SendgridHelperTest do
+  use ExUnit.Case
+
+  import Bamboo.SendgridHelper
+
+  @template_id "80509523-83de-42b6-a2bf-54b7513bd2aa"
+
+  setup do
+    {:ok, email: Bamboo.Email.new_email}
+  end
+
+  test "(with_template) adds the correct template", %{email: email} do
+    email = email |> with_template(@template_id)
+    assert email.private["x-smtpapi"] == %{
+                                            "filters" => %{
+                                              "templates" => %{
+                                                "settings" => %{
+                                                  "enable" => 1,
+                                                  "template_id" => @template_id
+                                                }
+                                              }
+                                            }
+                                          }
+  end
+
+  test "(with_template) raises on non-UUID `template_id`", %{email: email} do
+    assert_raise FunctionClauseError, fn ->
```

Comment:
Rather than raising a FunctionClauseError, what do you think of adding a catch all function that raises with a clear message?

Something like:

```
raise "expected the template ID to be a UUID that is 36 characters long, instead got: #{uuid}"
```